mediagrains/gsf.py

```diff
@@ -623,6 +625,9 @@ def grains(self, skip_data=False, load_lazily=False):
                     with GSFBlock(self.file_data, want_tag="gbhd", raise_on_wrong_tag=True) as gbhd_block:
                         meta = self._decode_gbhd(gbhd_block)
 
+                    if local_ids is not None and local_id not in local_ids:
```

Comment:
Would it work if this was directly after `local_id = grai_block.read_uint(2)` above? That would avoid reading the grain metadata (although that might not make any difference in most cases with the I/O block size.)